packages/web-components/src/search/product-search-overlay.ts

```diff
@@ -79,31 +86,45 @@ export class ProductSearchOverlay extends LitElement {
         }
 
         switch (event.key) {
-        case 'ArrowUp':
-            event.preventDefault();
-            this.selectedIndex = Math.max(this.selectedIndex - 1, 0);
-            break;
-        case 'ArrowDown':
-            event.preventDefault();
-            this.selectedIndex = Math.min(this.selectedIndex + 1, this.results.length - 1);
-            break;
-        case 'Tab':
-            event.preventDefault();
-            this.selectedIndex = Math.min(this.selectedIndex + 1, this.results.length - 1);
-            break;
-        case 'Enter':
-            event.preventDefault();
-            this.handleActionOnResult(this.results[this.selectedIndex]);
-            break;
+            case 'ArrowUp':
+                event.preventDefault();
+                this.selectedIndex = Math.max(this.selectedIndex - 1, 0);
+                break;
+            case 'ArrowDown':
+                event.preventDefault();
+                this.selectedIndex = Math.min(this.selectedIndex + 1, this.results.length - 1);
+                break;
+            case 'Tab':
+                event.preventDefault();
+                this.selectedIndex = Math.min(this.selectedIndex + 1, this.results.length - 1);
+                break;
+            case 'Enter':
+                event.preventDefault();
+                this.handleActionOnResult(this.results[this.selectedIndex]);
+                break;
         }
     }
 
-    handleActionOnResult(result: SearchResult) {
+    redirectToSearchPage() {
+        if (!this.searchPageUrl) return;
+        const url = new URL(this.searchPageUrl, window.location.href);
+        url.searchParams.set('rw-term', this.term);
+        window.location.href = url.toString();
+    }
+
+    handleActionOnResult(result?: SearchResult) {
+        if (result?.showAllResults === true) {
+            if (!this.searchPageUrl) return;
+            this.redirectToSearchPage();
+            return;
```

Comment:
Why is this no longer using `else if` seems a bit more brittle when you now need to remember to add a `return`